### PR TITLE
HWKAGENT-90 register the agent feed under all the tenant IDs from all endpoints

### PR DESCRIPTION
**Pull request by jmazzitelli**

### CODE REVIEW

File: hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/service/MonitorService.java [hunk 1]
```diff
@@ -903,40 +904,63 @@ private void registerFeed(String tenantId) throws Exception {
...
-                    // should we throw an error here or just use the feed ID we were given?
-                    log.debugf("Using feed ID [%s]; make sure the agent doesn't lose its data file", feed.getId());
+            // because we can support multiple tenants, we need to register our feed under all tenants
+            List<AbstractEndpointConfiguration> endpoints = new ArrayList<>();
+            endpoints.addAll(configuration.getDmrConfiguration().getEndpoints().values());
+            endpoints.addAll(configuration.getJmxConfiguration().getEndpoints().values());
+            endpoints.addAll(configuration.getPlatformConfiguration().getEndpoints().values());
+            endpoints.addAll(configuration.getPrometheusConfiguration().getEndpoints().values());
+
+            List<String> tenantIds = new ArrayList<String>();
```
Comment (jshaughn) [inline hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/service/MonitorService.java:914]: Set<String>?

File: hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/service/MonitorService.java [hunk 2]
```diff
@@ -903,40 +904,63 @@ private void registerFeed(String tenantId) throws Exception {
...
+            tenantIds.add(configuration.getStorageAdapter().getTenantId()); // always register agent's global tenant ID
+            for (AbstractEndpointConfiguration endpoint : endpoints) {
+                String tenantId = endpoint.getTenantId();
```
Comment (jshaughn) [inline hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/service/MonitorService.java:917]: Does this return a proper default of 'hawkular'?

**Comment by jshaughn** [comment]
Other than the comments, LGTM

File: hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/service/MonitorService.java (hunk 2 above)
Comment (jmazzitelli) [inline hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/service/MonitorService.java:917]: yes, the setting of the default happens earlier (WildFly API will actually return us the default value if one isn't defined - happens earlier)

File: hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/service/MonitorService.java (hunk 1 above)
Comment (jmazzitelli) [inline hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/service/MonitorService.java:914]: doesn't matter. List, Set. whateve. Its not used other than here in this local code so it doesn't matter.

Comment (jshaughn) [inline hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/service/MonitorService.java:914]: OK, I just figured why make multiple calls if 10 managed servers all used the same tenantId.  if that's not an issue then go ahead and merge.

**Pull request merged**